### PR TITLE
@l2succes => Update to /collect2 app

### PR DESCRIPTION
**Pull request by mzikherman**
- uses generated types for props
- uses `... on Viewer` vs `...on Query` (i think the former is more idiomatic)

This got filtering all working!

Next step, new slider component. 😎

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 3629c4dfc2e615e36c9bccf3788c04d2660aa93e

https://deploy-preview-1212--artsy-reaction.netlify.com

File: src/Apps/Collect/Components/ArtworkGrid/CollectArtworkFilterRefetch.tsx [hunk 1]
```diff
@@ -72,57 +74,62 @@ export class CollectArtworkFilterRefetch extends Component<
...
-        aggregations {
-          slice
-          counts {
-            name
-            id
-          }
+  {
+    viewer: graphql`
+      fragment CollectArtworkFilterRefetch_viewer on Viewer
+        @argumentDefinitions(
```
Comment (l2succes) [inline src/Apps/Collect/Components/ArtworkGrid/CollectArtworkFilterRefetch.tsx]: I'm still a bit confused as to where in the relay container tree we need to include argument definitions.
cc @mzikherman @alloy

Comment (mzikherman) [reply src/Apps/Collect/Components/ArtworkGrid/CollectArtworkFilterRefetch.tsx]: Yea, the GraphQL spec says all of these declared variables are global, and can be used by all fragments. So you would only need to list it at the top of your query renderer: https://github.com/artsy/reaction/pull/1212/files#diff-36e1f38238a06214205cd08ccc4a670eR12 .

I believe `@arguments` and `@argumentDefinitions` are Relay-added directives (the generated queries do _not_ contain these), and without them I've gotten Relay compiler issues.

So I _think_ (~90% sure) that it's basically a Relay thing/quirk that you need to keep defining/passing them down like this. But I haven't played around with it too much recently.

Comment (alloy) [reply src/Apps/Collect/Components/ArtworkGrid/CollectArtworkFilterRefetch.tsx]: Correct, these are Relay specific directives that allow fragments to define variables, even though normally variables only get defined at the root of your operation. It’s akin to components having a props interface and passing those props along vs using global variables, both work but you lose visibility into them when going down that route. Visibility in this case means e.g. that Relay knows if variables are unused and probably more things.

I’d invite you to read the relevant parts of the Relay source to gain a full understanding, possibly extend the documentation on the topic, and thus sharing this knowledge with us all.

**Pull request merged**